### PR TITLE
Fix lint warnings

### PR DESCRIPTION
**Pull request by artonge**
This PR fixes most of the lint warnings displayed when building the frontend. This is mostly typings licence name fixes.

It also adds the following to `.eslintrc.js` to improve typings resolution from `node_modules` packages.

```
settings.jsdoc.mode: 'typescript',
```

The only remaining warnings are deprecation warnings.

### CODE REVIEW

**Comment by nickvergessen** [comment]
Conflicting files
dist/files-main.js.map
dist/files-personal-settings.js.map
dist/settings-apps-settings-apps.js
dist/settings-apps-settings-apps.js.map
dist/settings-users-settings-users.js
dist/settings-users-settings-users.js.map
dist/user-status-modal-user-status-modal.js
dist/user-status-modal-user-status-modal.js.map

**Review by PVince81 — APPROVED** [review]
:+1:

File: apps/accessibility/src/accessibilityoca.js [hunk 1]
```diff
@@ -4,7 +4,7 @@
  * @author Jan C. Borchardt <hey@jancborchardt.net>
  * @author John Molakvoæ <skjnldsv@protonmail.com>
  *
- * @license GNU AGPL version 3 or any later version
+ * @license GPL-3.0-or-later
```
Comment (MichaIng) [inline apps/accessibility/src/accessibilityoca.js:7]: Isn't this wrong as Nextcloud is indeed AGPL, not GPL?

File: dist/settings-apps-settings-apps.js [hunk 2]
```diff
@@ -3610,4 +3610,8 @@ render._withStripped = true
 /***/ })
 
 }]);
-//# sourceMappingURL=settings-apps-settings-apps.js.map?v=a5f2375a4af256153b26
\ No newline at end of file
+<<<<<<< HEAD
+//# sourceMappingURL=settings-apps-settings-apps.js.map?v=a5f2375a4af256153b26
+=======
+//# sourceMappingURL=settings-apps-settings-apps.js.map?v=a2e2fa5f03a6f10d3519
+>>>>>>> 5f2027f1e1 (Typing corrections)
```
Comment (MichaIng) [inline dist/settings-apps-settings-apps.js]: Conflict not resolved correctly.

File: apps/accessibility/src/accessibilityoca.js [hunk 3]
```diff
@@ -4,7 +4,7 @@
  * @author Jan C. Borchardt <hey@jancborchardt.net>
  * @author John Molakvoæ <skjnldsv@protonmail.com>
  *
- * @license GNU AGPL version 3 or any later version
+ * @license  GPL-3.0-or-later
```
[the remark this answers is not in the document]
Comment (MichaIng) [reply apps/accessibility/src/accessibilityoca.js]: The doubled space is still present in a few other cases.

File: apps/accessibility/src/accessibilityoca.js (hunk 1 above)
Comment (PVince81) [reply apps/accessibility/src/accessibilityoca.js:7]: good point!

Comment (tcitworld) [reply apps/accessibility/src/accessibilityoca.js:7]: Even though, you can't change the licence like that.
```suggestion
 * @license AGPL-3.0-or-later
```

Comment (MichaIng) [reply apps/accessibility/src/accessibilityoca.js:7]: The AGPL explicitly applies additional statements to ensure that e.g. hosting providers need to make changes to the source public as well: https://github.com/nextcloud/03a0f4c/blob/master/COPYING#L42-L48

An important addition for Nextcloud which is indeed offered a lot by hosting providers.

Comment (PVince81) [reply apps/accessibility/src/accessibilityoca.js:7]: in any case, this ticket is just about making the linter happy, not changing the license :sweat_smile: 

so fixing the identifier to say AGPL makes sense

File: dist/settings-apps-settings-apps.js (hunk 2 above)
Comment (artonge) [reply dist/settings-apps-settings-apps.js]: Will be fixed by: https://github.com/nextcloud/server/pull/30605

File: apps/accessibility/src/accessibilityoca.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (artonge) [reply apps/accessibility/src/accessibilityoca.js]: Good now

File: apps/accessibility/src/accessibilityoca.js (hunk 1 above)
Comment (artonge) [reply apps/accessibility/src/accessibilityoca.js]: Reseted to AGPL

File: dist/settings-apps-settings-apps.js (hunk 2 above)
Comment (artonge) [reply dist/settings-apps-settings-apps.js]: Manually cleared the dist folder before building. Good now.

**Comment by MichaIng** [comment]
Some files from `dist/` and files sharing app are removed, I wonder why this didn't cause a webpack diff/CI failure on the base branch 🤔. Otherwise LGTM, nice cleanup and standardisation 👍.

**Review by MichaIng — APPROVED** [review]

**Pull request merged**